File: segundo/1er cuatri/PDOO/ejercicios_clase/practica2_ejercicios_clase/entregas_ejercicios/entrega_16-10/practica2_ejercicios_clase_ruby/objeto.rb
# frozen_string_literal: true

class Objeto
  def initialize(nombre, tipo, ataque, defensa)
    @nombre = nombre
    @tipo = tipo
    @ataque = ataque
    @defensa = defensa
  end

  def to_s
    "Objeto #{@nombre}, tipo: #{@tipo}, ataque: #{@ataque}, defensa : #{@defensa})"
  end
end

File: segundo/1er cuatri/PDOO/ejercicios_clase/practica2_ejercicios_clase/src/practica2_ejercicios_clase/Objeto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package practica2_ejercicios_clase;

/**
 *
 * @author alvarogz
 */
public class Objeto {
    private String nombre;
    private TipoObjeto tipo;
    private float ataque;
    private float defensa;

    public Objeto(String nombre, TipoObjeto tipo, float ataque, float defensa) {
        this.nombre = nombre;
        this.tipo = tipo;
        this.ataque = ataque;
        this.defensa = defensa;
    }
    
    @Override
    public String toString() {
        return "Objeto: " + nombre + ", tipo: " + tipo + ", ataque: " + ataque + ", defensa: " + defensa + "\n";
    }
}
